core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -52,11 +52,22 @@ class NetworkAnalyzer {
   static getSummary(values) {
     values.sort((a, b) => a - b);
 
+    let median;
+    if (values.length <= 1) {
+      median = values[0];
```

Comment:
nit: the 1 case is handled fine by the odd case, and length 0 is guaranteed to be undefined. This makes it more explicit. Reaching into `values[0]` when it's empty just to get undefined is kinda confusing.

```suggestion
    if (values.length === 0) {
      median = undefined;
```

Reply:
Your code suggestion made TS unhappy, so I threw an error for `values.length === 0` which should never happen with real data but plenty of unit tests fail with that. So, I reverted.